### ISSUE TITLE
Support PRs that get merged with upstream/master 

### ISSUE THREAD

**Issue by wilzbach** [codecov/support#360]
At the `dlang` organization [we merge the PR into `upstream/{master,stable}`](https://github.com/dlang/phobos/blob/master/circleci.sh#L68).

For example for [this PR](https://github.com/dlang/phobos/pull/5036) this would atm generate this merge commit:

```
Parent: 395ae88ac75ee0c03f6475ad4a5b073cfaf4e084 (Merge pull request #5146 from JackStouffer/isInfinite)
Parent: 95d6d895faf12db426a8d00fe6477dbacaf07fda (Removed GC allocations from std.container.array)
Branch: master
Follows: v2.073.1, v2.073.1-b2
Precedes: 

    Merge branch 'pr/5036'
```

However now the generated `.lst` files from test suite are different and thus this results in
- line-shifted coverage annotation for diffs
- invalid `codecov/project` report (any other commit in master might have made this effect)

A better example for this is this PR (https://github.com/dlang/phobos/pull/5141) where no code was changed at all.

Proposed changes
- allow the user to supply the `parent_ref` via the bash uploader (i.e. the ref that CodeCov uses to compare the overall project coverage)

I am not sure how the line-shifts are handled ideally - maybe a line-matching logic could help?

For the example above there in theory exist these two files:
- withot merge: http://pastebin.com/hDB6jXWF
- with merge upstream/master: http://pastebin.com/MyCuGEwz (this one gets send to CodeCov)

Now removing the coverage annotation from them `cut -c9-` and running a simple `git diff` leads to a changeset (http://pastebin.com/pz0GQuDz) with which it should be possible to do a map the lines correctly.

**Comment by stevepeak** [comment]
@wilzbach thank you for the feature request. 

I have a quick question before diving into this. What value(s) would you set `parent_ref` to? Perhaps this is something we can automatically detect.

There are other customers asking about comparing commits. Perhaps they are related to your question.

**Comment by wilzbach** [comment]
> Perhaps this is something we can automatically detect.

Yep, that would be even better. Getting the list of all parents is quite easy:

```
git show --pretty=%P HEAD
git rev-list --parents -n1 HEAD # this also includes the current rev
```

But then it gets a bit tricky as you could also be testing a "normal merge commit", so for us being able to specify `parent_ref` manually would already be pretty awesome.

FWIW for a simple two-way merge, `git rev-list` would look like this:

```
git rev-list --parents -n1 HEAD
<current-sha> <target_ref - the ref in upstream/master> <base_ref - the branch>
```

However there can also be [octopus merges](https://www.destroyallsoftware.com/blog/2017/the-biggest-and-weirdest-commits-in-linux-kernel-git-history).

**Comment by stevepeak** [comment]
I believe issue #363 is a fix. Can you review that issue and request the beta on it. Thanks!
